src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-form/rgw-storage-class-form.component.html

```diff
@@ -1,356 +1,361 @@
-    </legend>
-    <div class="form-item form-item-append"
-         cdsRow>
-      <div cdsCol>
-        <!-- Zone Group -->
-        <cds-select
-          label="Zone Group Name"
-          i18n-label
-          formControlName="zonegroup"
-          id="zonegroup"
-          [invalid]="
-            storageClassForm.controls.zonegroup.invalid && storageClassForm.controls.zonegroup.dirty
-          "
-          (change)="onZonegroupChange()"
-          [invalidText]="zonegroupError"
-        >
-          <option *ngFor="let zonegrp of zonegroupNames"
-                  [value]="zonegrp.name"
-                  [selected]="zonegrp.name === storageClassForm.getValue('zonegroup')"
-                  i18n>
-            {{ zonegrp.name }}
-          </option>
-        </cds-select>
-        <ng-template #zonegroupError>
-          <span
-            class="invalid-feedback"
-            *ngIf="storageClassForm.showError('zonegroup', formDir, 'required')"
-            i18n
-            >This field is required.</span
-          >
-        </ng-template>
+  <ng-container *cdFormLoading="loading">
+    <form name="storageClassForm"
+          #formDir="ngForm"
+          [formGroup]="storageClassForm"
+          novalidate>
+      <div i18n="form title"
+           class="form-header">
+        {{ action | titlecase }} {{ resource | upperFirst }}
       </div>
-      <div cdsCol>
-        <!-- Placement Target -->
-        <cds-select
-          label="Placement Target"
-          i18n-label
-          formControlName="placement_target"
-          id="placement_target"
-          [invalid]="
-            storageClassForm.controls.placement_target.invalid &&
-            storageClassForm.controls.placement_target.dirty
-          "
-          [invalidText]="placementError"
-        >
-        <option [value]=""
-                i18n> --Select-- </option>
-        <option *ngFor="let placementTarget of placementTargets"
-                [value]="placementTarget"
-                [selected]="placementTarget === storageClassForm.getValue('placement_target')"
-                i18n>
-            {{ placementTarget }}
-        </option>
-        </cds-select>
-        <ng-template #placementError>
-          <span
-            class="invalid-feedback"
-            *ngIf="storageClassForm.showError('placement_target', formDir, 'required')"
-            i18n
-            >This field is required.</span
+      <legend>
+        <cd-help-text i18n>
+          All fields are required, except where marked optional.
+        </cd-help-text>
+      </legend>
+      <div class="form-item form-item-append"
+           cdsRow>
+        <div cdsCol>
+          <!-- Zone Group -->
+          <cds-select
+            label="Zone Group Name"
+            i18n-label
+            formControlName="zonegroup"
+            id="zonegroup"
+            [invalid]="
+              storageClassForm.controls.zonegroup.invalid && storageClassForm.controls.zonegroup.dirty
+            "
```

Comment:
Same comment for other places as well. The helper function `showError()` checks for same if you see here `cd-form-group.ts`
```
  /**
   * Indicates errors of the control in templates
   */
  showError(controlName: string, form: NgForm, errorName?: string): boolean {
    const control = this.get(controlName);
    return (
      (form?.submitted || control.dirty) &&
      (errorName ? control.hasError(errorName) : control.invalid)
    );
  }
  ```
  So why not use this ?